src/fabric_cicd/_parameter/_utils.py

```diff
@@ -10,17 +10,120 @@
+        # Check if it's valid and return the matched value as the find_value, if found
+        try:
+            regex = re.compile(find_value)
+            match = re.search(regex, file_content)
+            if match:
+                return match.group(1)
+
+        except re.error as e:
+            msg = f"Invalid regex pattern: {find_value}"
+            raise ValueError(msg) from e
+
+    # Otherwise, return the find_value as is
+    return find_value
+
+
+def extract_replace_value(workspace_obj: FabricWorkspace, replace_value: str) -> str:
+    """Extracts the replace_value and sets the value. Processes the replace_value if a valid variable is provided."""
+    # If $workspace variable, return the workspace ID value
+    if replace_value == "$workspace.id":
+        return workspace_obj.workspace_id
+
+    # If $items variable, return the item attribute value, if found
+    # Expected format: $items['type']['name'].attribute
+    # Support any fabric ci-cd Item object string attributes, such as type, name, guid
+    if replace_value.startswith("$items"):
+        return _extract_item_attribute(workspace_obj, replace_value)
+
+    # Otherwise, return the replace_value as is
+    return replace_value
+
+
+def _extract_item_attribute(workspace_obj: FabricWorkspace, variable: str) -> str:
+    """Extracts the item attribute value from the replace_value $items variable."""
+    try:
+        # Split the variable into parts (item and attribute)
+        var_parts = variable[6:].split(".", 1)
+        if len(var_parts) < 2:
+            logger.warning(f"Invalid $items syntax (missing attribute part): {variable}")
+            return variable
+
+        item = var_parts[0].strip()
+        attribute = var_parts[1].strip()
+
+        # Extract the item type and name strings using a regex pattern
+        match = re.search(constants.ITEM_VARIABLE_REGEX, item)
+        if not match or len(match.groups()) != 2:
+            logger.warning(f"Invalid $items syntax (invalid pattern): {item}")
+            return variable
+
+        item_type, item_name = match.groups()
+
+        # Validate items exist in the workspace
+        if item_type not in workspace_obj.deployed_items:
+            logger.warning(f"Item type '{item_type}' not found in deployed items")
+            return variable
+
+        if item_name not in workspace_obj.deployed_items[item_type]:
+            logger.warning(f"Item '{item_name}' not found in type '{item_type}'")
+            return variable
+
+        # Get the item object and look for the provided attribute
+        item_obj = workspace_obj.deployed_items[item_type][item_name]
+        attr_lookup = {name.lower(): name for name in dir(item_obj)}
+        attr_key = attribute.lower()
+
+        if attr_key not in attr_lookup:
+            logger.warning(f"Attribute '{attribute}' not found in {item_type}/{item_name}")
+            return variable
+
+        # Get the attribute value using the case-insensitive lookup
```

Comment:
If item type or item name is incorrect or not found then fail the deployment

call out case sensitivity